### PR TITLE
Get refreshed mcapp, gdns from etcd in actionHandlers

### PR DESCRIPTION
**Pull request by mrajashree**
To address: https://github.com/rancher/rancher/issues/18852

### CODE REVIEW

File: pkg/api/customization/globaldns/action_globaldns.go [hunk 1]
```diff
@@ -118,12 +120,37 @@ func (w *Wrapper) removeProjects(gDNS *v3.GlobalDNS, request *types.APIContext,
...
+		Steps:    7,
+	}
+
+	err := wait.ExponentialBackoff(backoff, func() (bool, error) {
+		gDNSlatest, err := w.GlobalDNSes.GetNamespaced(gDNS.Namespace, gDNS.Name, v1.GetOptions{})
+		if err != nil {
+			if !apierrors.IsNotFound(err) {
+				return false, err
+			}
+			return false, nil
```
Comment (kinarashah) [inline pkg/api/customization/globaldns/action_globaldns.go]: Why do we return nil if object is not found? If this is just called by action handler, maybe we can just return the msg?

File: pkg/api/customization/globaldns/action_globaldns.go [hunk 2]
```diff
@@ -118,12 +120,37 @@ func (w *Wrapper) removeProjects(gDNS *v3.GlobalDNS, request *types.APIContext,
...
+		}
+		gDNSlatest.Spec.ProjectNames = targetProjects
```
Comment (kinarashah) [inline pkg/api/customization/globaldns/action_globaldns.go]: Not sure but maybe we need to deepcopy before update?

File: pkg/api/customization/globaldns/action_globaldns.go (hunk 1 above)
Comment (mrajashree) [reply pkg/api/customization/globaldns/action_globaldns.go]: Yeah I guess that makes sense. I borrowed this from an existing piece of code using ExponentialBackoff, but it the action handler can't find the obj on Get then we can return that error

**Comment by mrajashree** [comment]
Will add integration test for this later, since I'm not sure how to add tests for actions addProject/removeProject
cc @alena1108

**Comment by alena1108** [comment]
LGTM

**Pull request merged**